Find a functionally equivalent code:android.app.backup.BackupAgent.onRestoreFile:COMMENT
Method Modifier: protected   hidden      
Comment:/**
 * Only specialized platform agents should overload this entry point to support
 * restores to crazy non-app locations.
 * @hide
 */

@@ -23,28 +23,30 @@
                 mode = -1;
             }
         }
+    } else if (domain.equals(FullBackup.NO_BACKUP_TREE_TOKEN)) {
+        basePath = getNoBackupFilesDir().getCanonicalPath();
     } else {
         // Not a supported location
         Log.i(TAG, "Unrecognized domain " + domain);

